src/Command/Branch/BranchChangelogCommand.php

```diff
@@ -25,16 +27,31 @@ protected function configure()
     {
         $this
             ->setName('branch:changelog')
-            ->setDescription('Reports what got fixed or closed since last release on current branch')
+            ->setDescription('Reports what got fixed or closed since last release on the branch')
+            ->addArgument(
+                'branch',
+                InputArgument::OPTIONAL,
+                'Branch to look for tags, when none is given the current branch is used'
+            )
+            ->addOption(
+                'search',
+                's',
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Regex pattern to use for searching', ['/#(?P<id>[0-9]+)/i']
+            )
             ->setHelp(
                 <<<EOF
-Reports what got fixed or closed since last release on current branch.
-reference: http://www.lornajane.net/posts/2014/github-powered-changelog-scripts
-
-The <info>%command.name%</info> command :
+Reports what got fixed or closed since the last release on current branch.
 
     <info>$ gush %command.name%</info>
 
+This command will search in all the commits in the given branch (that were made after the last tag)
+and tries to extract the issue numbers from the message, to only match a precise pattern use the
+<comment>--search</comment> option to specify one or multiple regex-patterns (with delimiters and flags).
+
+Say your issues are prefix with "JIRA-" or "DC-", to match them use the following:
+
+    <info>$ gush %command.name% --search="{JIRA-(?P<id>[0-9]+)}i" --search="{DC-(?P<id>[0-9]+)}i"</info>
```

Comment:
This command will search all the commits in the given branch (that were made after the last tag)
and will try to extract the issue numbers from the message. To only match a precise pattern, use the
<comment>--search</comment> option to specify one or multiple regex-patterns (with delimiters and flags).

For example:
If your issues are prefixed with "JIRA-" or "DC-", use the following: